java-lib/src/main/java/com/wavefront/ingester/JsonParser.java

```diff
@@ -0,0 +1,46 @@
+package com.wavefront.ingester;
+
+import wavefront.report.Annotation;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Map;
+
+public class JsonParser {
+    private final Map<String, Object> input;
+    private final String DELIMITER = "_";
+    private final List<String> ignoreFlatten;
+
+    public JsonParser(@Nonnull Map<String, Object> input, List<String> ignoreFlatten) {
+        this.input = input;
+        this.ignoreFlatten = ignoreFlatten;
+    }
+
+    public void flattenJson(List<Annotation> annotations) {
+        for (Map.Entry<String, Object> tagKV : input.entrySet()) {
+            recursiveHelper(tagKV.getKey(), tagKV.getValue(), annotations);
+        }
+    }
+
+    public void recursiveHelper(String key, Object value, List<Annotation> annotations) {
```

Comment:
nit: names which end in Helper are not typically not recommended esp. for public methods because too vague (it adds complexity in the form of requiring people to read the code instead of understanding the functions purpose.

I would make this private and use naming like flattenJsonInternal